src/qibo/transpiler/router.py

```diff
@@ -174,34 +174,57 @@ class CircuitMap:
     def __init__(
         self,
         initial_layout: dict,
-        circuit: Circuit,
+        circuit: Optional[Circuit] = None,
```

Comment:
@csookim do you agree that the circuit is not really optional or am I missing something?

Reply:
Ah I see, then I believe it would be better to move the circuit check right after the temporary one:
``` python
        if self._temporary:  # if temporary circuit, no need to store the blocks
            return
        elif circuit is None:
            raise_error(ValueError, "Circuit must be provided for a non temporary `CircuitMap`.")
        self._nqubits = circuit.nqubits  # number of qubits
```

Reply:
I updated it.